src/schema/journey.js

```diff
@@ -1,4 +1,75 @@
-
+exports.typeDefs = `
+  type TLJourney @beehiveTable(table_name: "tl_journeys", pk_column: "tl_journey_id") {
+    # teacher leader
+    teacher: Person! @beehiveRelation(target_type_name: "Person")
+    # current state within the TL Journey
+    stage: [TLJourneyStage!] @beehiveRelation(target_type_name: "TLJourneyStage")
+    # TLJourney Checklist
+    # checklist: [TLJourneyChecklist!]
+    state: JourneyState
+    state_reason: String
```

Comment:
what is `state_reason` for?

Reply:
As specified, "Reason" is a string field independent of a given JourneyState. In practice, field is intended to capture the reason a startup journey is PAUSED or ABANDONED.

Reply:
This is what I was thinking we'd want to use the idea of state change events to track the state change with it's reason and date. if we did that then there would be a list of state_change_events on the journey and state would be derived from that list rather than explicitly set. Beehive doesn't yet support derived fields like that yet however.

Reply:
Gotcha, opting for the event based approach, removed `state_reason`.